bin/build-plugin-zip.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+# Exit if any command fails.
+set -e
+
+# Change to the expected directory.
+cd "$(dirname "$0")"
+cd ..
+
+# Enable nicer messaging for build status.
+BLUE_BOLD='\033[1;34m';
+GREEN_BOLD='\033[1;32m';
+RED_BOLD='\033[1;31m';
+YELLOW_BOLD='\033[1;33m';
+COLOR_RESET='\033[0m';
+error () {
+	echo -e "\n${RED_BOLD}$1${COLOR_RESET}\n"
+}
+status () {
+	echo -e "\n${BLUE_BOLD}$1${COLOR_RESET}\n"
+}
+success () {
+	echo -e "\n${GREEN_BOLD}$1${COLOR_RESET}\n"
+}
+warning () {
+	echo -e "\n${YELLOW_BOLD}$1${COLOR_RESET}\n"
+}
+
+status "Time to release AMP ⚡️"
+
+if [ -z "$NO_CHECKS" ]; then
+	# Make sure there are no changes in the working tree. Release builds should be
+	# traceable to a particular commit and reliably reproducible.
+	changed=
+	if ! git diff --exit-code > /dev/null; then
+		changed="file(s) modified"
+	elif ! git diff --cached --exit-code > /dev/null; then
+		changed="file(s) staged"
+	fi
+	if [ -n "$changed" ]; then
+		git status
+		error "ERROR: Cannot build plugin zip with dirty working tree. ☝️
+		Commit your changes and try again."
+		exit 1
+	fi
+
+	# Do a dry run of the repository reset. Prompting the user for a list of all
+	# files that will be removed should prevent them from losing important files!
+	status "Resetting the repository to pristine condition. ✨"
+	to_clean=$(git clean -xdf --dry-run)
+	if [ -n "$to_clean" ]; then
+		echo "$to_clean"
+		warning "🚨 About to delete everything above! Is this okay? 🚨"
+		echo -n "[y]es/[N]o: "
+		read -r answer
+		if [ "$answer" != "${answer#[Yy]}" ]; then
+			# Remove ignored files to reset repository to pristine condition. Previous
+			# test ensures that changed files abort the plugin build.
+			status "Cleaning working directory... 🛀"
+			git clean -xdf
+		else
+			error "Fair enough; aborting. Tidy up your repo and try again. 🙂"
+			exit 1
+		fi
+	fi
+fi
+
+# Run the build.
+status "Installing dependencies... 📦"
+composer install
```

Comment:
Is this needed because `grunt.task.run( 'shell:composer_install' );` is being done during build?

Reply:
Yes, it is needed right now because things are a bit too intertwined. The grunt task `readme` calls the a tool from the `xwp/wp-dev-lib` PHP development dependency:

https://github.com/ampproject/amp-wp/blob/1.2.2/Gruntfile.js#L26-L27

I didn't want to rewrite half of the build pipeline in this PR. That's one of the tings the `Note` at the end of the PR message is referring to.
But let me know if you think I should do a bigger change in one go.